scout/server/blueprints/cases/controllers.py

```diff
@@ -390,19 +390,28 @@ def case(store, institute_obj, case_obj):
         case_obj["default_genes"], limit_genes
     )
 
+    # Fetch secondary findings in one db call:
+    case_causative_gene_filter = list(set(limit_genes + limit_genes_default_panels))
+    matching_causatives = store.case_matching_causatives(
+        case_obj=case_obj, limit_genes=case_causative_gene_filter
+    )
+
+    # Filter secondary findings:
+    other_causatives = []
+    other_causatives_in_default_panels = []
+    for causative in matching_causatives:
+        hgnc_ids = [gene.get("hgnc_id") for gene in causative.get("genes", [])]
+        # Sort into approriate list based if overlap with filter lists:
+        if set(hgnc_ids) & set(limit_genes):
+            other_causatives.append(causative)
+        if set(hgnc_ids) & set(limit_genes_default_panels):
+            other_causatives_in_default_panels.append(causative)
+
     data = {
         "institute": institute_obj,
         "case": case_obj,
-        "other_causatives": [
-            var
-            for var in store.case_matching_causatives(case_obj=case_obj, limit_genes=limit_genes)
-        ],
-        "default_other_causatives": [
-            var
-            for var in store.case_matching_causatives(
-                case_obj=case_obj, limit_genes=limit_genes_default_panels
-            )
-        ],
+        "other_causatives": other_causatives,
```

Comment:
```suggestion
        "other_causatives": [causative for causative in matching_causatives if set(causative.get("genes", [])) & set(limit_genes)]
```

Reply:
It's the same thing, but perhaps you could avoid a few lines of code by using this syntax instead?

Reply:
My idea with building `other_causatives` and `default_other_causatives` in the same for-loop was to only have to iterate once over the `matching_causative` cursor.

If it's not a bad idea then I can do a `matching_causatives = list(matching_causatives)` and iterate over it twice!

Reply:
I don't think it would take so much time to du the iteration twice. It won't be the thing that slows down the loading of the page at least. You can also consider to move the lines of code above 

```
 # Limit secondary findings according to institute settings
    limit_genes = store.safe_genes_filter(institute_obj["_id"])

    limit_genes_default_panels = _limit_genes_on_default_panels(
        case_obj["default_genes"], limit_genes
    )

    # Fetch secondary findings in one db call:
    case_causative_gene_filter = list(set(limit_genes + limit_genes_default_panels))
    matching_causatives = store.case_matching_causatives(
        case_obj=case_obj, limit_genes=case_causative_gene_filter
    )

    # Filter secondary findings:
    other_causatives = []
    other_causatives_in_default_panels = []
    for causative in matching_causatives:
        hgnc_ids = [gene.get("hgnc_id") for gene in causative.get("genes", [])]
        # Check for overlap:
        if set(hgnc_ids) & set(limit_genes):
            other_causatives.append(causative)
        if set(hgnc_ids) & set(limit_genes_default_panels):
            other_causatives_in_default_panels.append(causative)
```

Into a separate function that might return a tuple (other_causatives, other_causatives_in_default_panels)

That way the code would be more easy to test with a unit test..